Friend/firmware/firmware_v1.0/src/transport.c

```diff
@@ -65,7 +68,119 @@ static struct bt_gatt_attr dfu_service_attr[] = {
 };
 
 static struct bt_gatt_service dfu_service = BT_GATT_SERVICE(dfu_service_attr);
+//Acceleration data
+//this code activates the onboard accelerometer. some cute ideas may include shaking the necklace to color strobe
+//
+typedef struct sensors{
+
+	struct sensor_value a_x;
+	struct sensor_value a_y;
+	struct sensor_value a_z;
+    struct sensor_value g_x;
+    struct sensor_value g_y;
+    struct sensor_value g_z;
+
+};
+static struct sensors mega_sensor;
+static struct device *lsm6dsl_dev;
+//Arbritrary uuid, feel free to change
+static struct bt_uuid_128 accel_uuid = BT_UUID_INIT_128(BT_UUID_128_ENCODE(0x00000000,0x0000,0x1000,0x7450,0xBE2E44B06B00));
+static struct bt_uuid_128 accel_uuid_x = BT_UUID_INIT_128(BT_UUID_128_ENCODE(0x00000001,0x0000,0x1000,0x7450,0xBE2E44B06B00));
+
+static void accel_ccc_config_changed_handler(const struct bt_gatt_attr *attr, uint16_t value);
+static ssize_t accel_data_read_characteristic(struct bt_conn *conn, const struct bt_gatt_attr *attr, void *buf, uint16_t len, uint16_t offset);
+
+static struct bt_gatt_attr accel_service_attr[] = {
+    BT_GATT_PRIMARY_SERVICE(&accel_uuid),//primary description
+    BT_GATT_CHARACTERISTIC(&accel_uuid_x.uuid, BT_GATT_CHRC_READ | BT_GATT_CHRC_NOTIFY, BT_GATT_PERM_READ, accel_data_read_characteristic, NULL, NULL),//data type
+    BT_GATT_CCC(accel_ccc_config_changed_handler, BT_GATT_PERM_READ | BT_GATT_PERM_WRITE),//scheduler
+};
+static struct bt_gatt_service accel_service = BT_GATT_SERVICE(accel_service_attr);
+
+static ssize_t accel_data_read_characteristic(struct bt_conn *conn, const struct bt_gatt_attr *attr, void *buf, uint16_t len, uint16_t offset)
+{
+    printk("audio_data_read_characteristic\n");
+    int axis_mode = 6; //3 for accel, 6 for (also) gyro
+    return bt_gatt_attr_read(conn, attr, buf, len, offset, &axis_mode, sizeof(axis_mode));
+}
+
+
+#define ACCEL_REFRESH_INTERVAL 500 // 0.5 seconds
```

Comment:
Can we bump this to 100ms? (as long as it doesn't interfere with the audio and video data streams)

Reply:
Sure. I believe the main intention was for pendant gestures, so if we need to go back to it we can.